src/conversions/array.rs

```diff
@@ -0,0 +1,72 @@
+use crate::{exceptions, types::PySequence, FromPyObject, PyAny, PyResult, PyTryFrom};
+
+// Helper to safely create arrays since the standard library doesn't
+// provide one yet. Shouldn't be necessary in the future.
+#[cfg(min_const_generics)]
+struct ArrayGuard<T, const N: usize> {
+    dst: *mut T,
+    initialized: usize,
+}
+
+#[cfg(min_const_generics)]
+impl<T, const N: usize> Drop for ArrayGuard<T, N> {
+    fn drop(&mut self) {
```

Comment:
Based on the coverage, it looks like this code never gets exercised.

As it's pretty fiddly `unsafe` code, we should add a test for it. I guess we could call `try_create_array` with a function that generates, and confirm that the drop code modified some counter as expected.